### PR TITLE
Refactor mock engine support with strategies

### PR DESCRIPTION
**Pull request by xzhayon**
This is currently supporting PHPUnit and Prophecy.

### CODE REVIEW

**Review by Giuffre — CHANGES_REQUESTED** [review]
Ok, this work is impressive. Now we provide three strategies to generate mock objects. If you agree now we should fix de documentation and the test executors to running with all available mocking engines.

**Comment by xzhayon** [comment]
Sure! I have to complete the work on stub return values, and we should also think if we want to provide specific traits for each strategy.

**Comment by Giuffre** [comment]
Please, @xzhavilla review my updates on README.md. I think we should now push the fact that we support multiple generator engines. Do you agree?

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ Being such a simple project, **Moka** can be integrated in an already existing t
 
 ### `mock(string $fqcn, string $alias = null): MockProxy`
 
-Creates (if not existing already) a proxy containing a PHPUnit mock object for the class identified by `$fqcn` and optionally assigns an `$alias` to it.
+Creates (if not existing already) a proxy containing a mock object according to selected strategy for the class identified by `$fqcn` and optionally assigns an `$alias` to it.
```
Comment (Jean85) [inline README.md]: To THE selected strategy...

File: tests/Strategy/Prophecy/NoPriorityTokenTest.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
+declare(strict_types=1);
+
+namespace Strategy\Prophecy;
```
Comment (Jean85) [inline tests/Strategy/Prophecy/NoPriorityTokenTest.php]: Wrong namespace?

**Review by Jean85 — APPROVED** [review]
All good, just 2 minor fixes..

**Review by Giuffre — APPROVED** [review]

**Pull request merged**